crates/typos-cli/src/file.rs

```diff
@@ -127,12 +129,85 @@ impl FileChecker for FixTypos {
-                    let new_path = path.with_file_name(new_name);
-                    std::fs::rename(path, new_path)?;
+                    fix_file_name(path, &file_name.to_owned(), fixes)?;
+                }
+            }
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct AskFixTypos;
+
+impl FileChecker for AskFixTypos {
+    fn check_file(
+        &self,
+        path: &std::path::Path,
+        explicit: bool,
+        policy: &crate::policy::Policy<'_, '_, '_>,
+        reporter: &dyn report::Report,
+    ) -> Result<(), std::io::Error> {
+        if policy.check_files {
+            let (buffer, content_type) = read_file(path, reporter)?;
+            let bc = buffer.clone();
+            if !explicit && !policy.binary && content_type.is_binary() {
+                let msg = report::BinaryFile { path };
+                reporter.report(msg.into())?;
+            } else {
+                let mut fixes = Vec::new();
+
+                let mut accum_line_num = AccumulateLineNum::new();
+                for typo in check_bytes(&bc, policy) {
+                    let line_num = accum_line_num.line_num(&buffer, typo.byte_offset);
+                    let (line, line_offset) = extract_line(&buffer, typo.byte_offset);
+                    let msg = report::Typo {
+                        context: Some(report::FileContext { path, line_num }.into()),
+                        buffer: std::borrow::Cow::Borrowed(line),
+                        byte_offset: line_offset,
+                        typo: typo.typo.as_ref(),
+                        corrections: typo.corrections.clone(),
+                    };
+                    reporter.report(msg.into())?;
+
+                    if let Some(correction_index) = select_fix(&typo) {
+                        fixes.push((typo, correction_index));
+                    }
+
+                    println!("\n");
+                }
+
+                if !fixes.is_empty() || path == std::path::Path::new("-") {
+                    let buffer = fix_buffer_with_correction_index(buffer, fixes.into_iter());
+                    write_file(path, content_type, buffer, reporter)?;
+                }
+            }
+        }
+
+        if policy.check_filenames {
+            if let Some(file_name) = path.file_name().and_then(|s| s.to_str()) {
+                let mut fixes = Vec::new();
+
+                for typo in check_str(file_name, policy) {
+                    let msg = report::Typo {
+                        context: Some(report::PathContext { path }.into()),
+                        buffer: std::borrow::Cow::Borrowed(file_name.as_bytes()),
+                        byte_offset: typo.byte_offset,
+                        typo: typo.typo.as_ref(),
+                        corrections: typo.corrections.clone(),
+                    };
+                    reporter.report(msg.into())?;
+
+                    if let Some(correction_index) = select_fix(&typo) {
```

Comment:
We're only supposed to report things that aren't fixed

Reply:
"not fixed" means typos that only have one possible correction? Shouldn't we prompt whether the user wants to apply that single possible fix?

Or do you mean running `reporter.report(msg.into())?`? That part however displays the suggested fix to the user, doesn't it?

Reply:
reporting is meant for use with `--format`.  It seems you are trying to take advantage of the output formatting for outputting the part of the prompt.  Technically, this will break down for json output but people probably shouldn't use this for progammatic output.

We should at least have a `// HACK:` comment explaining the abuse of reporting for the sake of prompting the user and that technically we should only report when not correcting.

Reply:
> It seems you are trying to take advantage of the output formatting for outputting the part of the prompt. 
>
Oh well, I see, I didn't notice it's possible to specify other output format than the `long` format.